src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -658,6 +658,11 @@ private void AddTypeKind(INamedTypeSymbol symbol)
                 {
                     switch (symbol.TypeKind)
                     {
+                        case TypeKind.Class when !symbol.GetMembers(WellKnownMemberNames.CloneMethodName).IsEmpty:
```

Comment:
>!symbol.GetMembers(WellKnownMemberNames.CloneMethodName).IsEmpty [](start = 49, length = 64)

I think we should duplicate the filter condition from SynthesizedRecordClone.FindValidCloneMethod and leave a comment there to keep this place in sync in case of future changes to the logic.

Reply:
I'm okay to do the more complete check, but would prefer not to duplicate logic. I'm pushing a change to leverage `FindValidCloneMethod`. Let me know if okay